### DISCUSSION TITLE
Chart/Graph Libraries that integrate with MDX

### DISCUSSION THREAD

**Discussion by Mochael** [mdx-js/mdx#2337] [Q&A]
Hello,

Anybody use any libraries for adding pretty charts/graphs in your MDX files? I've been trying to embed some charts in my MDX files using various chart libraries (plotly, Nivo, Chartjs) with no luck

**Comment by ChristianMurphy** [comment]
Welcome @Mochael!
MDX is Markdown and JSX, there aren't MDX _specific_ libraries because they aren't needed.
You can use any charting library your JSX system supports.
Most likely that means a React charting library, but you could also use Svelte, Vue, etc which also have their own respective libraries.
If you are looking for a list of React charting libraries you can find one at https://github.com/brillout/awesome-react-components#chart

If you have a question about a charting library, most have support forums which can offer guidance on how to configure and use them.
If you believe the issue is in the MDX integration _specifically_, please aim to follow the support guidelines https://mdxjs.com/community/support/ and be specific about what your goal is and what you've tried so other can offer concrete recommendations.

**Reply by Mochael** [reply]
Thanks for the reply Christian! I was eventually able to get a chart rendering after I:
- Moved the chart logic to be a child component of the component that I reference in the MDX
- Added 'use-client' to the parent component that I'm importing into the MDX file
- Passed in parent component into the MDXProvider's MDXComponents
